sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java

```diff
@@ -570,7 +581,18 @@ public String getAccountName() {
      * @return A {@code String} representing all SAS query parameters.
      */
     public String generateAccountSas(AccountSasSignatureValues accountSasSignatureValues) {
+        checkAccess();
         return new AccountSasImplUtil(accountSasSignatureValues)
             .generateSas(SasImplUtils.extractSharedKeyCredential(getHttpPipeline()));
     }
+
+    /**
+     * Checks if service client was built with credentials.
+     */
+    private void checkAccess() {
```

Comment:
I think somehow having the word anonymous in the method name will make it more self descriptive.

Reply:
I'll rename it to throwOnAnonymousAccess